daemon/oci_windows.go

```diff
@@ -25,11 +25,31 @@ func (daemon *Daemon) createSpec(c *container.Container) (*specs.Spec, error) {
 	// In base spec
 	s.Hostname = c.FullHostname()
 
+	if err := daemon.setupSecretDir(c); err != nil {
+		return nil, err
+	}
+
 	// In s.Mounts
 	mounts, err := daemon.setupMounts(c)
 	if err != nil {
 		return nil, err
 	}
+
+	if !c.HasBeenStartedBefore && len(c.SecretReferences) > 0 {
```

Comment:
Something like "Ensure that we do not attempt to mount the filesystem of a Hyper-V container that has previously been started to stop an attack vector."